pkg/target/http.go

```diff
@@ -362,25 +392,78 @@ func (ht *HTTPTarget) Write(messages []*models.Message) (*models.TargetWriteResu
 				continue
 			}
 			defer resp.Body.Close()
-			if resp.StatusCode >= 200 && resp.StatusCode < 300 {
+			responseBody, err := io.ReadAll(resp.Body)
+
+			if err != nil {
+				failed = append(failed, goodMsgs...)
```

Comment:
Interesting one - if we have successfully sent the data, but we encounter a problem in reading the response body, what should happen?

Reply:
Ooops, I think we should read response body later, only if we get non success so no 2XX status code. 

Unless....we expect API returning `200 OK` and response body `{errorMessage: "Failed request. Please don't ack data"}` 😁

Reply:
You joke but that's not actually far from what Braze actually does do sometimes 😬 

Agree it should simply come after we've checked response code for success :)